Index: configure.in
===================================================================
RCS file: /cvsroot/xchat/xchat2/configure.in,v
retrieving revision 1.85
diff -u -r1.85 configure.in
--- configure.in	10 Sep 2005 04:09:43 -0000	1.85
+++ configure.in	10 Sep 2005 22:02:31 -0000
@@ -2,7 +2,7 @@
 
 AC_INIT(src)
 AM_CONFIG_HEADER(config.h)
-AM_INIT_AUTOMAKE(xchat, 2.4.5)
+AM_INIT_AUTOMAKE(xchat-gnome, 0.6svn)
 AM_MAINTAINER_MODE
 AC_PROG_CC
 AC_PROG_CPP
@@ -62,6 +62,10 @@
 [  --enable-memdebug       enable use of Memory Debug (default: no)],
         memdebug=$enableval, memdebug=no)
 
+AC_ARG_ENABLE(profiling,
+[  --enable-profiling      turn on profiling support (default: no)],
+        profiling=$enableval, profiling=no)
+
 AC_ARG_ENABLE(socks,
 [  --enable-socks          link with SOCKS5 library (default: no)],
         socks=$enableval, socks=no)
@@ -86,9 +90,9 @@
 [  --enable-textfe         build the text frontend (default: no)],
         textfe=$enableval, textfe=no)
 
-dnl AC_ARG_ENABLE(gnome,
-dnl [  --disable-gnome         disable use of gnome],
-dnl        gnome=$enableval, gnome=yes)
+AC_ARG_ENABLE(gnomefe,
+[  --disable-gnomefe       disable the gnome frontend],
+       gnomefe=$enableval, gnomefe=yes)
 
 dnl AC_ARG_ENABLE(zvt,
 dnl [  --disable-zvt           disable zvt/shelltab feature],
@@ -114,6 +118,14 @@
 [  --enable-tcl[=PATH]       directory with Tcl config file: tclConfig.sh],
         tcl=$enableval, tcl=yes)
 
+AC_ARG_ENABLE(url_scraper,
+[  --disable-url_scraper,  don't build the url scraper plugin],
+		urlscraper=$enableval, urlscraper=yes)
+
+AC_ARG_ENABLE(notification,
+[  --disable-notification,  don't build the notification plugin],
+	  notification=$enableval, notification=yes)
+
 AC_ARG_ENABLE(plugin,
 [  --disable-plugin        disable plugin support],
         plugin=$enableval, plugin=yes)
@@ -126,6 +138,10 @@
 [  --enable-shm            enable use of XShm for fast tinting (default: no)],
         shm=$enableval, shm=no)
 
+AC_ARG_ENABLE(libsexy,
+[  --disable-libsexy       disable use of libsexy],
+        libsexy=$enableval, libsexy=yes)
+
 dnl *********************************************************************
 dnl ** DEBUG ************************************************************
 dnl *********************************************************************
@@ -160,7 +176,6 @@
 fi
 
 if test "$gtkfe" != yes; then
-	gnome=no
 	zvt=no
 	COMMON_LIBS="$GLIB_LIBS"
 	COMMON_CFLAGS="$GLIB_CFLAGS"
@@ -173,22 +188,49 @@
 GUI_LIBS="$GUI_LIBS $GTK_LIBS"
 GUI_CFLAGS="$GUI_CFLAGS $GTK_CFLAGS"
 
-gnome=no
-#if test "$gnome" = yes; then
-#	AC_PATH_PROG(pkgconfigpath, pkg-config)
-#	AC_MSG_CHECKING(Gnome2 compile flags)
-#	GNOME_CFLAGS="`$pkgconfigpath libgnome-2.0 --cflags 2>/dev/null`"
-#	if test "_$GNOME_CFLAGS" = _ ; then
-#		gnome=no
-#		AC_MSG_RESULT([Gnome not found, building without it.])
-#	else
-#		GNOME_VER="`$pkgconfigpath libgnome-2.0 --modversion`"
-#		GUI_LIBS="$GUI_LIBS `$pkgconfigpath libgnome-2.0 --libs`"
-#		GUI_CFLAGS="$GUI_CFLAGS $GNOME_CFLAGS"
-#		AC_DEFINE(USE_GNOME)
-#		AC_MSG_RESULT(ok)
-#	fi
-#fi
+AM_PATH_GTK_2_0(2.6.0, havenewgtk=yes, havenewgtk=no)
+
+if test "$havenewgtk" != yes; then
+	gnomefe=no
+	echo
+	echo Cannot find GTK+ 2.6\! Not building GNOME FrontEnd.
+	echo
+fi
+
+if test "$gnomefe" = yes; then
+	AC_PATH_PROG(pkgconfigpath, pkg-config)
+	AC_MSG_CHECKING(Gnome2 compile flags)
+	GNOME_CFLAGS="`$pkgconfigpath libgnome-2.0 libgnomeui-2.0 libglade-2.0 gnome-vfs-2.0 gconf-2.0 --cflags 2>/dev/null`"
+	if test "_$GNOME_CFLAGS" = _ ; then
+		gnomefe=no
+		AC_MSG_RESULT([Gnome not found, building without it.])
+	else
+		GNOME_VER="`$pkgconfigpath libgnome-2.0 --modversion`"
+		GUI_LIBS="$GUI_LIBS `$pkgconfigpath libgnome-2.0 --libs`"
+		GUI_LIBS="$GUI_LIBS `$pkgconfigpath libgnomeui-2.0 --libs`";
+		GUI_LIBS="$GUI_LIBS `$pkgconfigpath libglade-2.0 --libs`";
+		GUI_LIBS="$GUI_LIBS `$pkgconfigpath gnome-vfs-2.0 --libs`";
+		GUI_LIBS="$GUI_LIBS `$pkgconfigpath gconf-2.0 --libs`";
+		GUI_CFLAGS="$GUI_CFLAGS $GNOME_CFLAGS"
+		AC_MSG_RESULT(ok)
+	fi
+
+fi
+
+# GConf
+AC_PATH_PROG(GCONFTOOL, gconftool-2, no)
+AM_GCONF_SOURCE_2
+
+# soft dependency on libsexy
+if test "$libsexy" = yes; then
+	LIBSEXY_CFLAGS="`$pkgconfigpath --cflags libsexy 2>/dev/null`"
+	if test "_$LIBSEXY_CFLAGS" != _ ; then
+		LIBSEXY_LIBS="`$pkgconfigpath --libs libsexy`";
+		LIBSEXY_CFLAGS="$LIBSEXY_CFLAGS -DHAVE_LIBSEXY"
+		AC_SUBST(LIBSEXY_CFLAGS)
+		AC_SUBST(LIBSEXY_LIBS)
+	fi
+fi
 
 dnl *********************************************************************
 dnl ** XFT **************************************************************
@@ -467,9 +509,12 @@
 AM_CONDITIONAL(USE_OPENSSL, test "x$openssl" = "xyes")
 AM_CONDITIONAL(DO_TEXT, test "x$textfe" = "xyes")
 AM_CONDITIONAL(DO_GTK, test "x$gtkfe" = "xyes")
+AM_CONDITIONAL(DO_GNOME, test "x$gnomefe" = "xyes")
 AM_CONDITIONAL(DO_PERL, test "x$perl" = "xyes")
 AM_CONDITIONAL(DO_PYTHON, test "x$python" = "xyes")
 AM_CONDITIONAL(DO_TCL, test "x$tcl" = "xyes")
+AM_CONDITIONAL(DO_URLSCRAPER, test "x$urlscraper" = "xyes")
+AM_CONDITIONAL(DO_NOTIFICATION, test "x$notification" = "xyes")
 AM_CONDITIONAL(DO_PLUGIN, test "x$plugin" = "xyes")
 
 dnl *********************************************************************
@@ -563,6 +608,10 @@
 	fi
 fi
 
+if test x$enable_profiling = xyes; then
+	CFLAGS="$CFLAGS -pg"
+fi
+
 dnl does this compiler support -Wno-pointer-sign ?
 AC_MSG_CHECKING([if gcc accepts -Wno-pointer-sign ])
 
@@ -699,12 +748,16 @@
 src/Makefile
 src/common/Makefile
 src/fe-text/Makefile
+src/fe-gnome/Makefile
+src/fe-gnome/data/Makefile
 src/fe-gtk/Makefile
 src/pixmaps/Makefile
 plugins/Makefile
 plugins/python/Makefile
 plugins/perl/Makefile
 plugins/tcl/Makefile
+plugins/url_scraper/Makefile
+plugins/notification/Makefile
 intl/Makefile
 po/Makefile.in
 ])
@@ -712,10 +765,14 @@
 echo
 echo xchat $VERSION
 echo
-echo Building GTK+ Interface .... : $gtkfe
-echo Building TEXT Interface .... : $textfe
+echo Building GTK+  Interface .... : $gtkfe
+echo Building TEXT  Interface .... : $textfe
+echo Building GNOME Interface .... : $gnomefe
 echo
-echo PLUGINS: Perl: $perl Python: $python TCL: $tcl
+echo PLUGINS:
+echo Perl: $perl\    Python: $python 
+echo TCL: $tcl\      Url Scraper: $urlscraper
+echo Notification: $notification
 echo
 echo mmx tinting ......... : $mmx
 echo XShm tinting ........ : $shm\	plugin interface .... : $plugin
Index: plugins/Makefile.am
===================================================================
RCS file: /cvsroot/xchat/xchat2/plugins/Makefile.am,v
retrieving revision 1.4
diff -u -r1.4 Makefile.am
--- plugins/Makefile.am	13 Jan 2003 07:29:36 -0000	1.4
+++ plugins/Makefile.am	10 Sep 2005 22:02:31 -0000
@@ -1,10 +1,10 @@
 # temp. solution while mailcheck & xdcc arn't in SUBDIRS
-EXTRA_DIST = plugin20.html Make.plugin plugin-conf.in \
-				mailcheck/mailcheck.c \
-				mailcheck/mailcheck-config.h.in \
-				mailcheck/Makefile.am \
-				xdcc/xdcc.c \
-				xdcc/Makefile.am
+EXTRA_DIST =		\
+	plugin20.html	\
+	Make.plugin	\
+	plugin-conf.in	\
+	xdcc/xdcc.c	\
+	xdcc/Makefile.am
 
 noinst_HEADERS = xchat-plugin.h
 
@@ -20,5 +20,13 @@
 tcldir = tcl
 endif
 
+if DO_URLSCRAPER
+urlscrprdir = url_scraper
+endif
+
+if DO_NOTIFICATION
+notificationdir = notification
+endif
+
 #SUBDIRS = . $(pythondir) $(perldir) mailcheck xdcc
-SUBDIRS = . $(pythondir) $(perldir) $(tcldir)
+SUBDIRS = . $(pythondir) $(perldir) $(tcldir) $(urlscrprdir) $(notificationdir)
Index: plugins/mailcheck/Makefile.am
===================================================================
RCS file: /cvsroot/xchat/xchat2/plugins/mailcheck/Makefile.am,v
retrieving revision 1.1
diff -u -r1.1 Makefile.am
--- plugins/mailcheck/Makefile.am	17 Dec 2002 15:36:39 -0000	1.1
+++ plugins/mailcheck/Makefile.am	10 Sep 2005 22:02:31 -0000
@@ -10,7 +10,6 @@
 
 # This file must be in the form PLUGIN-config.h.in, it can be empty but
 # must exist
-EXTRA_DIST = mailcheck-config.h.in
 
 # Remember to include this line in your Makefile.am
 include @top_srcdir@/plugins/Make.plugin
Index: plugins/perl/Makefile.am
===================================================================
RCS file: /cvsroot/xchat/xchat2/plugins/perl/Makefile.am,v
retrieving revision 1.10
diff -u -r1.10 Makefile.am
--- plugins/perl/Makefile.am	29 Mar 2005 05:11:50 -0000	1.10
+++ plugins/perl/Makefile.am	10 Sep 2005 22:02:31 -0000
@@ -1,5 +1,5 @@
 
-EXTRA_DIST=alt_completion.pl ignore.pl sample_script.pl xchat2-perldocs.html xchat2-perl.html generate_header Xchat.pm IRC.pm makefile.msc
+EXTRA_DIST=alt_completion.pl ignore.pl sample_script.pl xchat2-perldocs.html xchat2-perl.html generate_header Xchat.pm IRC.pm makefile.msc generate_header
 
 libdir = $(xchatlibdir)/plugins
 
Index: src/Makefile.am
===================================================================
RCS file: /cvsroot/xchat/xchat2/src/Makefile.am,v
retrieving revision 1.1.1.1
diff -u -r1.1.1.1 Makefile.am
--- src/Makefile.am	28 Nov 2002 10:39:00 -0000	1.1.1.1
+++ src/Makefile.am	10 Sep 2005 22:03:39 -0000
@@ -11,4 +11,8 @@
 gtk_fe = fe-gtk
 endif
 
-SUBDIRS = pixmaps common $(gtk_fe) $(text_fe)
+if DO_GNOME
+gnome_fe = fe-gnome
+endif
+
+SUBDIRS = pixmaps common $(gtk_fe) $(text_fe) $(gnome_fe)
Index: src/common/textevents.h
===================================================================
RCS file: /cvsroot/xchat/xchat2/src/common/textevents.h,v
retrieving revision 1.30
diff -u -r1.30 textevents.h
--- src/common/textevents.h	11 Jul 2005 07:03:53 -0000	1.30
+++ src/common/textevents.h	10 Sep 2005 22:03:39 -0000
@@ -3,214 +3,214 @@
 const struct text_event te[] = {
 
 {"Add Notify", pevt_generic_nick_help, 1, 
-N_("%C22*%O$t$1 added to notify list.")},
+N_("-%C10-%C11-%O$t$1 added to notify list.")},
 
 {"Ban List", pevt_banlist_help, 4, 
-N_("%C22*%O$t$1 Banlist:%C19 $4%C20 $2%C21 $3")},
+N_("-%C10-%C11-%O$t$1 Banlist: %C3$4 %C4$2 %C5$3%O")},
 
 {"Banned", pevt_generic_channel_help, 1, 
-N_("%C22*%O$tCannot join%C26 %B$1 %O(You are banned).")},
+N_("-%C10-%C11-%O$tCannot join%C11 %B$1 %O(You are banned).")},
 
 {"Beep", pevt_generic_none_help, 0, 
 ""},
 
 {"Change Nick", pevt_changenick_help, 2, 
-N_("%C22*%O$t$1 is now known as $2")},
+N_("-%C10-%C11-%O$t$1 is now known as $2")},
 
 {"Channel Action", pevt_chanaction_help, 2, 
-N_("%C18*$t$1%O $2")},
+N_("%C13*%O$t$1 $2%O")},
 
 {"Channel Action Hilight", pevt_chanaction_help, 2, 
-N_("%C21*%O$t%C21%B$1%O%C21 $2")},
+N_("%C13*%O$t%C8%B$1%B%O $2%O")},
 
 {"Channel Ban", pevt_chanban_help, 2, 
-N_("%C22*%O$t$1 sets ban on $2")},
+N_("-%C10-%C11-%O$t$1 sets ban on $2")},
 
 {"Channel Creation", pevt_chandate_help, 2, 
-N_("%C22*%O$tChannel $1 created on $2")},
+N_("-%C10-%C11-%O$tChannel $1 created on $2")},
 
 {"Channel DeHalfOp", pevt_chandehop_help, 2, 
-N_("%C22*%O$t%C26$1%O removes channel half-operator status from%C26 $2")},
+N_("-%C10-%C11-%O$t$1 removes channel half-operator status from $2")},
 
 {"Channel DeOp", pevt_chandeop_help, 2, 
-N_("%C22*%O$t%C26$1%O removes channel operator status from%C26 $2")},
+N_("-%C10-%C11-%O$t$1 removes channel operator status from $2")},
 
 {"Channel DeVoice", pevt_chandevoice_help, 2, 
-N_("%C22*%O$t%C26$1%O removes voice from%C26 $2")},
+N_("-%C10-%C11-%O$t$1 removes voice from $2")},
 
 {"Channel Exempt", pevt_chanexempt_help, 2, 
-N_("%C22*%O$t$1 sets exempt on $2")},
+N_("-%C10-%C11-%O$t$1 sets exempt on $2")},
 
 {"Channel Half-Operator", pevt_chanhop_help, 2, 
-N_("%C22*%O$t%C26$1%O gives channel half-operator status to%C26 $2")},
+N_("-%C10-%C11-%O$t$1 gives channel half-operator status to $2")},
 
 {"Channel INVITE", pevt_chaninvite_help, 2, 
-N_("%C22*%O$t$1 sets invite on $2")},
+N_("-%C10-%C11-%O$t$1 sets invite on $2")},
 
 {"Channel List", pevt_generic_none_help, 0, 
-N_("%UChannel          Users   Topic")},
+N_("%UChannel          Users   Topic%O")},
 
 {"Channel Message", pevt_chanmsg_help, 4, 
-N_("%C18%B%B$4$1%O%C18%O$t$2")},
+N_("%C2<%O$1%C2>%O$t$2%O")},
 
 {"Channel Mode Generic", pevt_chanmodegen_help, 4, 
-N_("%C22*%O$t$1 sets mode $2$3 $4")},
+N_("-%C10-%C11-%O$t$1 sets mode $2$3 $4")},
 
 {"Channel Modes", pevt_chanmodes_help, 2, 
-N_("%C22*%O$t%C22Channel $1 modes: $2")},
+N_("-%C10-%C11-%O$tChannel $1 modes: $2")},
 
 {"Channel Msg Hilight", pevt_chanmsg_help, 4, 
-N_("$4%C21%B$1%O%C21$t$2")},
+N_("%C2<%C8%B$1%B%C2>%O$t$2%O")},
 
 {"Channel Notice", pevt_channotice_help, 3, 
-N_("%C28-%C29$1/$2%C28-%O$t$3")},
+N_("%C12-%C13$1/$2%C12-%O$t$3%O")},
 
 {"Channel Operator", pevt_chanop_help, 2, 
-N_("%C22*%O$t%C26$1%O gives channel operator status to%C26 $2")},
+N_("-%C10-%C11-%O$t$1 gives channel operator status to $2")},
 
 {"Channel Remove Exempt", pevt_chanrmexempt_help, 2, 
-N_("%C22*%O$t$1 removes exempt on $2")},
+N_("-%C10-%C11-%O$t$1 removes exempt on $2")},
 
 {"Channel Remove Invite", pevt_chanrminvite_help, 2, 
-N_("%C22*%O$t$1 removes invite on $2")},
+N_("-%C10-%C11-%O$t$1 removes invite on $2")},
 
 {"Channel Remove Keyword", pevt_chanrmkey_help, 1, 
-N_("%C22*%O$t$1 removes channel keyword")},
+N_("-%C10-%C11-%O$t$1 removes channel keyword")},
 
 {"Channel Remove Limit", pevt_chanrmlimit_help, 1, 
-N_("%C22*%O$t$1 removes user limit")},
+N_("-%C10-%C11-%O$t$1 removes user limit")},
 
 {"Channel Set Key", pevt_chansetkey_help, 2, 
-N_("%C22*%O$t$1 sets channel keyword to $2")},
+N_("-%C10-%C11-%O$t$1 sets channel keyword to $2")},
 
 {"Channel Set Limit", pevt_chansetlimit_help, 2, 
-N_("%C22*%O$t$1 sets channel limit to $2")},
+N_("-%C10-%C11-%O$t$1 sets channel limit to $2")},
 
 {"Channel UnBan", pevt_chanunban_help, 2, 
-N_("%C22*%O$t$1 removes ban on $2")},
+N_("-%C10-%C11-%O$t$1 removes ban on $2")},
 
 {"Channel Voice", pevt_chanvoice_help, 2, 
-N_("%C22*%O$t%C26$1%O gives voice to%C26 $2")},
+N_("-%C10-%C11-%O$t$1 gives voice to $2")},
 
 {"Connected", pevt_generic_none_help, 0, 
-N_("%C22*%O$t%C22Connected. Now logging in...")},
+N_("-%C10-%C11-%O$tConnected. Now logging in..")},
 
 {"Connecting", pevt_connect_help, 3, 
-N_("%C22*%O$t%C22Connecting to $1 ($2) port $3%O...")},
+N_("-%C10-%C11-%O$tConnecting to %C11$1 %C14(%C11$2%C14)%C port %C11$3%C..")},
 
 {"Connection Failed", pevt_connfail_help, 1, 
-N_("%C21*%O$t%C21Connection failed. Error: $1")},
+N_("-%C10-%C11-%O$tConnection failed. Error: $1")},
 
 {"CTCP Generic", pevt_ctcpgen_help, 2, 
-N_("%C22*%O$tReceived a CTCP $1 from $2")},
+N_("-%C10-%C11-%O$tReceived a CTCP $1 from $2")},
 
 {"CTCP Generic to Channel", pevt_ctcpgenc_help, 3, 
-N_("%C22*%O$tReceived a CTCP $1 from $2 (to $3)")},
+N_("-%C10-%C11-%O$tReceived a CTCP $1 from $2 (to $3)")},
 
 {"CTCP Send", pevt_ctcpsend_help, 2, 
-N_("%C19>%O$1%C19<%O$tCTCP $2")},
+N_("%C3>%O$1%C3<%O$tCTCP $2%O")},
 
 {"CTCP Sound", pevt_ctcpsnd_help, 2, 
-N_("%C22*%O$tReceived a CTCP Sound $1 from $2")},
+N_("-%C10-%C11-%O$tReceived a CTCP Sound $1 from $2")},
 
 {"CTCP Sound to Channel", pevt_ctcpsnd_help, 3, 
 N_("%C22*%O$tReceived a CTCP Sound $1 from $2 (to $3)")},
 
 {"DCC CHAT Abort", pevt_dccchatabort_help, 1, 
-N_("%C22*%O$tDCC CHAT to %C26$1%O aborted.")},
+N_("-%C10-%C11-%O$tDCC CHAT to %C11$1%O aborted.")},
 
 {"DCC CHAT Connect", pevt_dccchatcon_help, 2, 
-N_("%C22*%O$tDCC CHAT connection established to %C26$1 %C30[%O$2%C30]")},
+N_("-%C10-%C11-%O$tDCC CHAT connection established to %C11$1 %C14[%O$2%C14]%O")},
 
 {"DCC CHAT Failed", pevt_dccchaterr_help, 4, 
-N_("%C22*%O$tDCC CHAT to %C26$1%O lost ($4).")},
+N_("-%C10-%C11-%O$tDCC CHAT failed. Connection to $1 %C14[%O$2:$3%C14]%O lost.")},
 
 {"DCC CHAT Offer", pevt_generic_nick_help, 1, 
-N_("%C22*%O$tReceived a DCC CHAT offer from $1")},
+N_("-%C10-%C11-%O$tReceived a DCC CHAT offer from $1")},
 
 {"DCC CHAT Offering", pevt_generic_nick_help, 1, 
-N_("%C22*%O$tOffering DCC CHAT to $1")},
+N_("-%C10-%C11-%O$tOffering DCC CHAT to $1")},
 
 {"DCC CHAT Reoffer", pevt_generic_nick_help, 1, 
-N_("%C22*%O$tAlready offering CHAT to $1")},
+N_("-%C10-%C11-%O$tAlready offering CHAT to $1")},
 
 {"DCC Conection Failed", pevt_dccconfail_help, 3, 
-N_("%C22*%O$tDCC $1 connect attempt to%C26 $2%O failed (err=$3).")},
+N_("-%C10-%C11-%O$tDCC $1 connect attempt to %C11$2%O failed (err=$3).")},
 
 {"DCC Generic Offer", pevt_dccgenericoffer_help, 2, 
-N_("%C22*%O$tReceived '$1%O' from $2")},
+N_("-%C10-%C11-%O$tReceived '$1%O' from $2")},
 
 {"DCC Header", pevt_generic_none_help, 0, 
 N_("%C24,18 Type  To/From    Status  Size    Pos     File         ")},
 
 {"DCC Malformed", pevt_malformed_help, 2, 
-N_("%C22*%O$tReceived a malformed DCC request from %C26$1%O.%010%C22*%O$tContents of packet: $2")},
+N_("-%C10-%C11-%O$tReceived a malformed DCC request from %C11$1%O.%010-%C10-%C11-%O$tContents of packet: $2")},
 
 {"DCC Offer", pevt_dccoffer_help, 3, 
-N_("%C22*%O$tOffering%C26 $1%O to%C26 $2")},
+N_("-%C10-%C11-%O$tOffering %C11$1 %Cto %C11$2%O")},
 
 {"DCC Offer Not Valid", pevt_generic_none_help, 0, 
-N_("%C22*%O$tNo such DCC offer.")},
+N_("-%C10-%C11-%O$tNo such DCC offer.")},
 
 {"DCC RECV Abort", pevt_dccfileabort_help, 2, 
-N_("%C22*%O$tDCC RECV%C26 $2%O to%C26 $1%O aborted.")},
+N_("-%C10-%C11-%O$tDCC RECV %C11$2%O to %C11$1%O aborted.")},
 
 {"DCC RECV Complete", pevt_dccrecvcomp_help, 4, 
-N_("%C22*%O$tDCC RECV%C26 $1%O from%C26 $3%O complete %C30[%C26$4%O cps%C30]%O.")},
+N_("-%C10-%C11-%O$tDCC RECV %C11$1%O from %C11$3%O complete %C14[%C11$4%O cps%C14]%O.")},
 
 {"DCC RECV Connect", pevt_dcccon_help, 3, 
-N_("%C22*%O$tDCC RECV connection established to%C26 $1 %C30[%O$2%C30]")},
+N_("-%C10-%C11-%O$tDCC RECV connection established to %C11$1 %C14[%O$2%C14]%O")},
 
 {"DCC RECV Failed", pevt_dccrecverr_help, 4, 
-N_("%C22*%O$tDCC RECV%C26 $1%O from%C26 $3%O failed ($4).")},
+N_("-%C10-%C11-%O$tDCC RECV $1 ($2) failed. Connection to $3 lost.")},
 
 {"DCC RECV File Open Error", pevt_generic_file_help, 2, 
-N_("%C22*%O$tDCC RECV: Cannot open $1 for writing ($2).")},
+N_("-%C10-%C11-%O$tDCC RECV: Cannot open $1 for writing ($2).")},
 
 {"DCC Rename", pevt_dccrename_help, 2, 
-N_("%C22*%O$tThe file%C26 $1%C already exists, saving it as%C26 $2%O instead.")},
+N_("-%C10-%C11-%O$tThe file %C11$1%C already exists, saving it as %C11$2%O instead.")},
 
 {"DCC RESUME Request", pevt_dccresumeoffer_help, 3, 
-N_("%C22*%O$t%C26$1 %Ohas requested to resume%C26 $2 %Cfrom%C26 $3%C.")},
+N_("-%C10-%C11-%O$t%C11$1 %Chas requested to resume %C11$2 %Cfrom %C11$3%C.")},
 
 {"DCC SEND Abort", pevt_dccfileabort_help, 2, 
-N_("%C22*%O$tDCC SEND%C26 $2%O to%C26 $1%O aborted.")},
+N_("-%C10-%C11-%O$tDCC SEND %C11$2%O to %C11$1%O aborted.")},
 
 {"DCC SEND Complete", pevt_dccsendcomp_help, 3, 
-N_("%C22*%O$tDCC SEND%C26 $1%O to%C26 $2%O complete %C30[%C26$3%O cps%C30]%O.")},
+N_("-%C10-%C11-%O$tDCC SEND %C11$1%O to %C11$2%O complete %C14[%C11$3%O cps%C14]%O.")},
 
 {"DCC SEND Connect", pevt_dcccon_help, 3, 
-N_("%C22*%O$tDCC SEND connection established to%C26 $1 %C30[%O$2%C30]")},
+N_("-%C10-%C11-%O$tDCC SEND connection established to %C11$1 %C14[%O$2%C14]%O")},
 
 {"DCC SEND Failed", pevt_dccsendfail_help, 3, 
-N_("%C22*%O$tDCC SEND%C26 $1%O to%C26 $2%O failed. $3")},
+N_("-%C10-%C11-%O$tDCC SEND %C11$1%O failed. Connection to %C11$2%O lost.")},
 
 {"DCC SEND Offer", pevt_dccsendoffer_help, 4, 
-N_("%C22*%O$t%C26$1 %Ohas offered%C26 $2 %O(%C26$3 %Obytes)")},
+N_("-%C10-%C11-%O$t%C11$1 %Chas offered %C11$2 %C(%C11$3 %Cbytes)")},
 
 {"DCC Stall", pevt_dccstall_help, 3, 
-N_("%C22*%O$tDCC $1%C26 $2 %Oto%C26 $3 %Cstalled - aborting.")},
+N_("-%C10-%C11-%O$tDCC $1 %C11$2 %Cto %C11$3 %Cstalled - aborting.")},
 
 {"DCC Timeout", pevt_dccstall_help, 3, 
-N_("%C22*%O$tDCC $1%C26 $2 %Oto%C26 $3 %Otimed out - aborting.")},
+N_("-%C10-%C11-%O$tDCC $1 %C11$2 %Cto %C11$3 %Ctimed out - aborting.")},
 
 {"Delete Notify", pevt_generic_nick_help, 1, 
-N_("%C22*%O$t$1 deleted from notify list.")},
+N_("-%C10-%C11-%O$t$1 deleted from notify list.")},
 
 {"Disconnected", pevt_discon_help, 1, 
-N_("%C22*%O$tDisconnected ($1).")},
+N_("-%C10-%C11-%O$tDisconnected ($1).")},
 
 {"Found IP", pevt_foundip_help, 1, 
-N_("%C22*%O$tFound your IP: [$1]")},
+N_("-%C10-%C11-%O$tFound your IP: [$1]")},
 
 {"Generic Message", pevt_genmsg_help, 2, 
 N_("$1$t$2")},
 
 {"Ignore Add", pevt_ignoreaddremove_help, 1, 
-N_("%O%C26$1%O added to ignore list.")},
+N_("%O%C11$1%O added to ignore list.")},
 
 {"Ignore Changed", pevt_ignoreaddremove_help, 1, 
-N_("Ignore on %C26$1%O changed.")},
+N_("Ignore on %C11$1%O changed.")},
 
 {"Ignore Footer", pevt_generic_none_help, 0, 
 N_("%C24,18                                                              ")},
@@ -219,194 +219,194 @@
 N_("%C24,18 Hostmask                  PRIV NOTI CHAN CTCP DCC  INVI UNIG ")},
 
 {"Ignore Remove", pevt_ignoreaddremove_help, 1, 
-N_("%O%C26$1%O removed from ignore list.")},
+N_("%O%C11$1%O removed from ignore list.")},
 
 {"Ignorelist Empty", pevt_generic_none_help, 0, 
 N_("  Ignore list is empty.")},
 
 {"Invite", pevt_generic_channel_help, 1, 
-N_("%C22*%O$tCannot join%C26 %B$1 %O(Channel is invite only).")},
+N_("-%C10-%C11-%O$tCannot join%C11 %B$1 %O(Channel is invite only).")},
 
 {"Invited", pevt_invited_help, 3, 
-N_("%C22*%O$tYou have been invited to%C26 $1%O by%C26 $2%C (%C26$3%C)")},
+N_("-%C10-%C11-%O$tYou have been invited to %C11$1%C by %C11$2%C (%C11$3%C)")},
 
 {"Join", pevt_join_help, 3, 
-N_("%C19*%O$t%C19%B$1 %B($3) has joined $2")},
+N_("-%C10-%C11>%O$t%B$1%B %C14(%C10$3%C14)%C has joined $2")},
 
 {"Keyword", pevt_generic_channel_help, 1, 
-N_("%C22*%O$tCannot join%C26 %B$1 %O(Requires keyword).")},
+N_("-%C10-%C11-%O$tCannot join%C11 %B$1 %O(Requires keyword).")},
 
 {"Kick", pevt_kick_help, 4, 
-N_("%C21*%O$t%C21$1 has kicked $2 from $3 ($4%O%C21)")},
+N_("<%C10-%C11-%O$t$1 has kicked $2 from $3 ($4%O)")},
 
 {"Killed", pevt_kill_help, 2, 
-N_("%C22*%O$tYou have been killed by $1 ($2%O%C22)")},
+N_("-%C10-%C11-%O$tYou have been killed by $1 ($2%O)")},
 
 {"Message Send", pevt_ctcpsend_help, 2, 
-N_("%C19>%O$1%C19<%O$t$2")},
+N_("%C3>%O$1%C3<%O$t$2%O")},
 
 {"Motd", pevt_servertext_help, 1, 
-N_("%C16*%O$t$1%O")},
+N_("-%C10-%C11-%O$t$1%O")},
 
 {"MOTD Skipped", pevt_generic_none_help, 0, 
-N_("%C22*%O$t%C22MOTD Skipped.")},
+N_("-%C10-%C11-%O$tMOTD Skipped.")},
 
 {"Nick Clash", pevt_nickclash_help, 2, 
-N_("%C22*%O$t$1 already in use. Retrying with $2...")},
+N_("-%C10-%C11-%O$t$1 already in use. Retrying with $2..")},
 
 {"Nick Failed", pevt_generic_none_help, 0, 
-N_("%C22*%O$tNickname already in use. Use /NICK to try another.")},
+N_("-%C10-%C11-%O$tNickname already in use. Use /NICK to try another.")},
 
 {"No DCC", pevt_generic_none_help, 0, 
-N_("%C22*%O$tNo such DCC.")},
+N_("-%C10-%C11-%O$tNo such DCC.")},
 
 {"No Running Process", pevt_generic_none_help, 0, 
-N_("%C22*%O$tNo process is currently running")},
+N_("-%C10-%C11-%O$tNo process is currently running")},
 
 {"Notice", pevt_notice_help, 2, 
-N_("%C28-%C29$1%C28-%O$t$2")},
+N_("%C12-%C13$1%C12-%O$t$2%O")},
 
 {"Notice Send", pevt_ctcpsend_help, 2, 
-N_("%C19>%O$1%C19<%O$t$2")},
+N_("%C3>%O$1%C3<%O$t$2%O")},
 
 {"Notify Empty", pevt_generic_none_help, 0, 
-N_("$tNotify list is empty.")},
+N_("-%C10-%C11-%O$tNotify list is empty.")},
 
 {"Notify Header", pevt_generic_none_help, 0, 
 N_("%C24,18 %B  Notify List                           ")},
 
 {"Notify Number", pevt_notifynumber_help, 1, 
-N_("%C22*%O$t$1 users in notify list.")},
+N_("-%C10-%C11-%O$t$1 users in notify list.")},
 
 {"Notify Offline", pevt_generic_nick_help, 2, 
-N_("%C22*%O$tNotify: $1 is offline ($2).")},
+N_("-%C10-%C11-%O$tNotify: $1 is offline ($2).")},
 
 {"Notify Online", pevt_generic_nick_help, 2, 
-N_("%C22*%O$tNotify: $1 is online ($2).")},
+N_("-%C10-%C11-%O$tNotify: $1 is online ($2).")},
 
 {"Open Dialog", pevt_generic_none_help, 0, 
 ""},
 
 {"Part", pevt_part_help, 3, 
-N_("%C23*%O$t%C23$1 (%O%C23$2) has left $3")},
+N_("<%C10-%C11-%O$t$1 %C14(%O$2%C14)%C has left $3")},
 
 {"Part with Reason", pevt_partreason_help, 4, 
-N_("%C23*%O$t%C23$1 (%O%C23$2) has left $3 (%O%C23%B%B$4%O%C23)")},
+N_("<%C10-%C11-%O$t$1 %C14(%O$2%C14)%C has left $3 %C14(%O$4%C14)%O")},
 
 {"Ping Reply", pevt_pingrep_help, 2, 
-N_("%C22*%O$tPing reply from $1: $2 second(s)")},
+N_("-%C10-%C11-%O$tPing reply from $1 : $2 second(s)")},
 
 {"Ping Timeout", pevt_pingtimeout_help, 1, 
-N_("%C22*%O$tNo ping reply for $1 seconds, disconnecting.")},
+N_("-%C10-%C11-%O$tNo ping reply for $1 seconds, disconnecting.")},
 
 {"Private Message", pevt_privmsg_help, 3, 
-N_("%C28*%C29$3$1%C28*$t%O$2")},
+N_("%C12*%C13$1%C12*$t%O$2%O")},
 
 {"Private Message to Dialog", pevt_dprivmsg_help, 3, 
-N_("%C18%B%B$3$1%O$t$2")},
+N_("%C2<%O$1%C2>%O$t$2%O")},
 
 {"Process Already Running", pevt_generic_none_help, 0, 
-N_("%C22*%O$tA process is already running")},
+N_("-%C10-%C11-%O$tA process is already running")},
 
 {"Quit", pevt_quit_help, 3, 
-N_("%C23*%O$t%C23$1 has quit (%O%C23%B%B$2%O%C23)")},
+N_("<%C10-%C11-%O$t$1 has quit %C14(%O$2%O%C14)%O")},
 
 {"Raw Modes", pevt_rawmodes_help, 2, 
-N_("%C22*%O$t$1 sets modes%B %C30[%O$2%B%C30]")},
+N_("-%C10-%C11-%O$t$1 sets modes%B %C14[%O$2%B%C14]%O")},
 
 {"Receive Wallops", pevt_dprivmsg_help, 2, 
-N_("%C28-%C29$1/Wallops%C28-%O$t$2")},
+N_("%C12-%C13$1/Wallops%C12-%O$t$2%O")},
 
 {"Resolving User", pevt_resolvinguser_help, 2, 
-N_("%C22*%O$tLooking up IP number for%C26 $1%O...")},
+N_("-%C10-%C11-%O$tLooking up IP number for%C11 $1%O..")},
 
 {"Server Connected", pevt_generic_none_help, 0, 
-N_("%C22*%O$t%C22Connected.")},
+N_("-%C10-%C11-%O$tConnected.")},
 
 {"Server Error", pevt_servererror_help, 1, 
-N_("%C22*%O$t$1")},
+N_("-%C10-%C11-%O$t$1%O")},
 
 {"Server Lookup", pevt_serverlookup_help, 1, 
-N_("%C22*%O$t%C22Looking up $1")},
+N_("-%C10-%C11-%O$tLooking up %C11$1%C..")},
 
 {"Server Notice", pevt_servertext_help, 2, 
-N_("%C22*%O$t$1")},
+N_("-%C10-%C11-%O$t$1%O")},
 
 {"Server Text", pevt_servertext_help, 2, 
-N_("%C22*%O$t$1")},
+N_("-%C10-%C11-%O$t$1%O")},
 
 {"Stop Connection", pevt_sconnect_help, 1, 
-N_("%C22*%O$tStopped previous connection attempt (pid=$1)")},
+N_("-%C10-%C11-%O$tStopped previous connection attempt (pid=$1)")},
 
 {"Topic", pevt_topic_help, 2, 
-N_("%C29*%O$t%C29Topic for $1%C %C29is: $2")},
+N_("-%C10-%C11-%O$tTopic for %C11$1%C is %C11$2%O")},
 
 {"Topic Change", pevt_newtopic_help, 3, 
-N_("%C22*%O$t$1 has changed the topic to: $2")},
+N_("-%C10-%C11-%O$t$1 has changed the topic to: $2%O")},
 
 {"Topic Creation", pevt_topicdate_help, 3, 
-N_("%C29*%O$t%C29Topic for $1%C %C29set by $2%C %C29at $3")},
+N_("-%C10-%C11-%O$tTopic for %C11$1%C set by %C11$2%C at %C11$3%O")},
 
 {"Unknown Host", pevt_generic_none_help, 0, 
-N_("%C22*%O$tUnknown host. Maybe you misspelled it?")},
+N_("-%C10-%C11-%O$tUnknown host. Maybe you misspelled it?")},
 
 {"User Limit", pevt_generic_channel_help, 1, 
-N_("%C22*%O$tCannot join%C26 %B$1 %O(User limit reached).")},
+N_("-%C10-%C11-%O$tCannot join%C11 %B$1 %O(User limit reached).")},
 
 {"Users On Channel", pevt_usersonchan_help, 2, 
-N_("%C22*%O$t%C26Users on $1:%C $2")},
+N_("-%C10-%C11-%O$t%C11Users on $1:%C $2")},
 
 {"WhoIs Authenticated", pevt_whoisauth_help, 3, 
 N_("%C22*%O$t%C28[%O$1%C28] %O$2%C27 $3")},
 
 {"WhoIs Away Line", pevt_whois5_help, 2, 
-N_("%C22*%O$t%C28[%O$1%C28] %Cis away %C30(%O$2%O%C30)")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12] %Cis away %C14(%O$2%O%C14)")},
 
 {"WhoIs Channel/Oper Line", pevt_whois2_help, 2, 
-N_("%C22*%O$t%C28[%O$1%C28]%O $2")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12]%C $2")},
 
 {"WhoIs End", pevt_whois6_help, 1, 
-N_("%C22*%O$t%C28[%O$1%C28] %OEnd of WHOIS list.")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12] %CEnd of WHOIS list.")},
 
 {"WhoIs Identified", pevt_whoisid_help, 2, 
-N_("%C22*%O$t%C28[%O$1%C28]%O $2")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12] %C$2")},
 
 {"WhoIs Idle Line", pevt_whois4_help, 2, 
-N_("%C22*%O$t%C28[%O$1%C28]%O idle%C26 $2")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12]%O idle %C11$2%O")},
 
 {"WhoIs Idle Line with Signon", pevt_whois4t_help, 3, 
-N_("%C22*%O$t%C28[%O$1%C28]%O idle%C26 $2%O, signon:%C26 $3")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12]%O idle %C11$2%O, signon: %C11$3%O")},
 
 {"WhoIs Name Line", pevt_whois1_help, 4, 
-N_("%C22*%O$t%C28[%O$1%C28] %C30(%O$2@$3%C30)%O: $4")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12] %C14(%O$2@$3%C14) %O: $4%O")},
 
 {"WhoIs Real Host", pevt_whoisrealhost_help, 4, 
-N_("%C22*%O$t%C28[%O$1%C28] %Oreal user@host%C27 $2%O, real IP%C27 $3")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12] %Oreal user@host %C11$2%O, real IP %C11$3%O")},
 
 {"WhoIs Server Line", pevt_whois3_help, 2, 
-N_("%C22*%O$t%C28[%O$1%C28]%O $2")},
+N_("-%C10-%C11-%O$t%C12[%O$1%C12]%O $2")},
 
 {"WhoIs Special", pevt_whoisid_help, 3, 
 N_("%C22*%O$t%C28[%O$1%C28]%O $2")},
 
-{"You Join", pevt_join_help, 3, 
-N_("%C19*%O$t%C19Now talking on $2")},
+{"You Join", pevt_ukick_help, 4, 
+N_("-%C10-%C11>%O$t%BYou%B are now talking on %C11$2%O")},
 
 {"You Kicked", pevt_ukick_help, 4, 
-N_("%C23*$tYou have been kicked from $2 by $3 ($4%O%C23)")},
+N_("-%C10-%C11-%O$tYou have been kicked from $2 by $3 ($4%O)")},
 
 {"You Part", pevt_part_help, 3, 
-N_("%C23*$tYou have left channel $3")},
+N_("-%C10-%C11-%O$tYou have left channel $3")},
 
 {"You Part with Reason", pevt_partreason_help, 4, 
-N_("%C23*$tYou have left channel $3 (%O%C23%B%B$4%O%C23)")},
+N_("-%C10-%C11-%O$tYou have left channel $3 %C14(%O$4%C14)%O")},
 
 {"Your Invitation", pevt_uinvite_help, 3, 
-N_("%C22*%O$tYou've invited%C26 $1%O to%C26 $2%O (%C26$3%O)")},
+N_("-%C10-%C11-%O$tYou're inviting %C11$1%C to %C11$2%C (%C11$3%C)")},
 
 {"Your Message", pevt_chanmsg_help, 4, 
-N_("%C31%B%B$4$1%O%C30$t$2")},
+N_("%C6<%O$1%C6>%O$t$2%O")},
 
 {"Your Nick Changing", pevt_uchangenick_help, 2, 
-N_("%C22*%O$tYou are now known as $2")},
+N_("-%C10-%C11-%O$tYou are now known as $2")},
 };
Index: src/common/textevents.in
===================================================================
RCS file: /cvsroot/xchat/xchat2/src/common/textevents.in,v
retrieving revision 1.29
diff -u -r1.29 textevents.in
--- src/common/textevents.in	11 Jul 2005 07:03:53 -0000	1.29
+++ src/common/textevents.in	10 Sep 2005 22:03:39 -0000
@@ -1,19 +1,19 @@
 Add Notify
 XP_TE_ADDNOTIFY
 pevt_generic_nick_help
-%C22*%O$t$1 added to notify list.
+-%C10-%C11-%O$t$1 added to notify list.
 1
 
 Ban List
 XP_TE_BANLIST
 pevt_banlist_help
-%C22*%O$t$1 Banlist:%C19 $4%C20 $2%C21 $3
+-%C10-%C11-%O$t$1 Banlist: %C3$4 %C4$2 %C5$3%O
 4
 
 Banned
 XP_TE_BANNED
 pevt_generic_channel_help
-%C22*%O$tCannot join%C26 %B$1 %O(You are banned).
+-%C10-%C11-%O$tCannot join%C11 %B$1 %O(You are banned).
 1
 
 Beep
@@ -25,199 +25,199 @@
 Change Nick
 XP_TE_CHANGENICK
 pevt_changenick_help
-%C22*%O$t$1 is now known as $2
+-%C10-%C11-%O$t$1 is now known as $2
 2
 
 Channel Action
 XP_TE_CHANACTION
 pevt_chanaction_help
-%C18*$t$1%O $2
+%C13*%O$t$1 $2%O
 2
 
 Channel Action Hilight
 XP_TE_HCHANACTION
 pevt_chanaction_help
-%C21*%O$t%C21%B$1%O%C21 $2
+%C13*%O$t%C8%B$1%B%O $2%O
 2
 
 Channel Ban
 XP_TE_CHANBAN
 pevt_chanban_help
-%C22*%O$t$1 sets ban on $2
+-%C10-%C11-%O$t$1 sets ban on $2
 2
 
 Channel Creation
 XP_TE_CHANDATE
 pevt_chandate_help
-%C22*%O$tChannel $1 created on $2
+-%C10-%C11-%O$tChannel $1 created on $2
 2
 
 Channel DeHalfOp
-XP_TE_CHANDEHOP
+XP_TE_CHAN
 pevt_chandehop_help
-%C22*%O$t%C26$1%O removes channel half-operator status from%C26 $2
+-%C10-%C11-%O$t$1 removes channel half-operator status from $2
 2
 
 Channel DeOp
 XP_TE_CHANDEOP
 pevt_chandeop_help
-%C22*%O$t%C26$1%O removes channel operator status from%C26 $2
+-%C10-%C11-%O$t$1 removes channel operator status from $2
 2
 
 Channel DeVoice
 XP_TE_CHANDEVOICE
 pevt_chandevoice_help
-%C22*%O$t%C26$1%O removes voice from%C26 $2
+-%C10-%C11-%O$t$1 removes voice from $2
 2
 
 Channel Exempt
-XP_TE_CHANEXEMPT
+XP_TE_CHANEXCEMPT
 pevt_chanexempt_help
-%C22*%O$t$1 sets exempt on $2
+-%C10-%C11-%O$t$1 sets exempt on $2
 2
 
 Channel Half-Operator
 XP_TE_CHANHOP
 pevt_chanhop_help
-%C22*%O$t%C26$1%O gives channel half-operator status to%C26 $2
+-%C10-%C11-%O$t$1 gives channel half-operator status to $2
 2
 
 Channel INVITE
 XP_TE_CHANINVITE
 pevt_chaninvite_help
-%C22*%O$t$1 sets invite on $2
+-%C10-%C11-%O$t$1 sets invite on $2
 2
 
 Channel List
 XP_TE_CHANLISTHEAD
 pevt_generic_none_help
-%UChannel          Users   Topic
+%UChannel          Users   Topic%O
 0
 
 Channel Message
 XP_TE_CHANMSG
 pevt_chanmsg_help
-%C18%B%B$4$1%O%C18%O$t$2
+%C2<%O$1%C2>%O$t$2%O
 4
 
 Channel Mode Generic
 XP_TE_CHANMODEGEN
 pevt_chanmodegen_help
-%C22*%O$t$1 sets mode $2$3 $4
+-%C10-%C11-%O$t$1 sets mode $2$3 $4
 4
 
 Channel Modes
 XP_TE_CHANMODES
 pevt_chanmodes_help
-%C22*%O$t%C22Channel $1 modes: $2
+-%C10-%C11-%O$tChannel $1 modes: $2
 2
 
 Channel Msg Hilight
-XP_TE_HCHANMSG
+XP_TE_CHANMSG
 pevt_chanmsg_help
-$4%C21%B$1%O%C21$t$2
+%C2<%C8%B$1%B%C2>%O$t$2%O
 4
 
 Channel Notice
 XP_TE_CHANNOTICE
 pevt_channotice_help
-%C28-%C29$1/$2%C28-%O$t$3
+%C12-%C13$1/$2%C12-%O$t$3%O
 3
 
 Channel Operator
 XP_TE_CHANOP
 pevt_chanop_help
-%C22*%O$t%C26$1%O gives channel operator status to%C26 $2
+-%C10-%C11-%O$t$1 gives channel operator status to $2
 2
 
 Channel Remove Exempt
 XP_TE_CHANRMEXEMPT
 pevt_chanrmexempt_help
-%C22*%O$t$1 removes exempt on $2
+-%C10-%C11-%O$t$1 removes exempt on $2
 2
 
 Channel Remove Invite
 XP_TE_CHANRMINVITE
 pevt_chanrminvite_help
-%C22*%O$t$1 removes invite on $2
+-%C10-%C11-%O$t$1 removes invite on $2
 2
 
 Channel Remove Keyword
 XP_TE_CHANRMKEY
 pevt_chanrmkey_help
-%C22*%O$t$1 removes channel keyword
+-%C10-%C11-%O$t$1 removes channel keyword
 1
 
 Channel Remove Limit
 XP_TE_CHANRMLIMIT
 pevt_chanrmlimit_help
-%C22*%O$t$1 removes user limit
+-%C10-%C11-%O$t$1 removes user limit
 1
 
 Channel Set Key
 XP_TE_CHANSETKEY
 pevt_chansetkey_help
-%C22*%O$t$1 sets channel keyword to $2
+-%C10-%C11-%O$t$1 sets channel keyword to $2
 2
 
 Channel Set Limit
 XP_TE_CHANSETLIMIT
 pevt_chansetlimit_help
-%C22*%O$t$1 sets channel limit to $2
+-%C10-%C11-%O$t$1 sets channel limit to $2
 2
 
 Channel UnBan
 XP_TE_CHANUNBAN
 pevt_chanunban_help
-%C22*%O$t$1 removes ban on $2
+-%C10-%C11-%O$t$1 removes ban on $2
 2
 
 Channel Voice
 XP_TE_CHANVOICE
 pevt_chanvoice_help
-%C22*%O$t%C26$1%O gives voice to%C26 $2
+-%C10-%C11-%O$t$1 gives voice to $2
 2
 
 Connected
 XP_TE_CONNECTED
 pevt_generic_none_help
-%C22*%O$t%C22Connected. Now logging in...
+-%C10-%C11-%O$tConnected. Now logging in..
 0
 
 Connecting
 XP_TE_CONNECT
 pevt_connect_help
-%C22*%O$t%C22Connecting to $1 ($2) port $3%O...
+-%C10-%C11-%O$tConnecting to %C11$1 %C14(%C11$2%C14)%C port %C11$3%C..
 3
 
 Connection Failed
 XP_TE_CONNFAIL
 pevt_connfail_help
-%C21*%O$t%C21Connection failed. Error: $1
+-%C10-%C11-%O$tConnection failed. Error: $1
 1
 
 CTCP Generic
 XP_TE_CTCPGEN
 pevt_ctcpgen_help
-%C22*%O$tReceived a CTCP $1 from $2
+-%C10-%C11-%O$tReceived a CTCP $1 from $2
 2
 
 CTCP Generic to Channel
 XP_TE_CTCPGENC
 pevt_ctcpgenc_help
-%C22*%O$tReceived a CTCP $1 from $2 (to $3)
+-%C10-%C11-%O$tReceived a CTCP $1 from $2 (to $3)
 3
 
 CTCP Send
 XP_TE_CTCPSEND
 pevt_ctcpsend_help
-%C19>%O$1%C19<%O$tCTCP $2
+%C3>%O$1%C3<%O$tCTCP $2%O
 2
 
 CTCP Sound
 XP_TE_CTCPSND
 pevt_ctcpsnd_help
-%C22*%O$tReceived a CTCP Sound $1 from $2
+-%C10-%C11-%O$tReceived a CTCP Sound $1 from $2
 2
 
 CTCP Sound to Channel
@@ -229,49 +229,49 @@
 DCC CHAT Abort
 XP_TE_DCCCHATABORT
 pevt_dccchatabort_help
-%C22*%O$tDCC CHAT to %C26$1%O aborted.
+-%C10-%C11-%O$tDCC CHAT to %C11$1%O aborted.
 1
 
 DCC CHAT Connect
 XP_TE_DCCCONCHAT
 pevt_dccchatcon_help
-%C22*%O$tDCC CHAT connection established to %C26$1 %C30[%O$2%C30]
+-%C10-%C11-%O$tDCC CHAT connection established to %C11$1 %C14[%O$2%C14]%O
 2
 
 DCC CHAT Failed
 XP_TE_DCCCHATF
 pevt_dccchaterr_help
-%C22*%O$tDCC CHAT to %C26$1%O lost ($4).
+-%C10-%C11-%O$tDCC CHAT failed. Connection to $1 %C14[%O$2:$3%C14]%O lost.
 4
 
 DCC CHAT Offer
 XP_TE_DCCCHATOFFER
 pevt_generic_nick_help
-%C22*%O$tReceived a DCC CHAT offer from $1
+-%C10-%C11-%O$tReceived a DCC CHAT offer from $1
 1
 
 DCC CHAT Offering
 XP_TE_DCCCHATOFFERING
 pevt_generic_nick_help
-%C22*%O$tOffering DCC CHAT to $1
+-%C10-%C11-%O$tOffering DCC CHAT to $1
 1
 
 DCC CHAT Reoffer
 XP_TE_DCCCHATREOFFER
 pevt_generic_nick_help
-%C22*%O$tAlready offering CHAT to $1
+-%C10-%C11-%O$tAlready offering CHAT to $1
 1
 
 DCC Conection Failed
 XP_TE_DCCCONFAIL
 pevt_dccconfail_help
-%C22*%O$tDCC $1 connect attempt to%C26 $2%O failed (err=$3).
+-%C10-%C11-%O$tDCC $1 connect attempt to %C11$2%O failed (err=$3).
 3
 
 DCC Generic Offer
 XP_TE_DCCGENERICOFFER
 pevt_dccgenericoffer_help
-%C22*%O$tReceived '$1%O' from $2
+-%C10-%C11-%O$tReceived '$1%O' from $2
 2
 
 DCC Header
@@ -283,121 +283,121 @@
 DCC Malformed
 XP_TE_MALFORMED
 pevt_malformed_help
-%C22*%O$tReceived a malformed DCC request from %C26$1%O.%010%C22*%O$tContents of packet: $2
+-%C10-%C11-%O$tReceived a malformed DCC request from %C11$1%O.%010-%C10-%C11-%O$tContents of packet: $2
 2
 
 DCC Offer
 XP_TE_DCCOFFER
 pevt_dccoffer_help
-%C22*%O$tOffering%C26 $1%O to%C26 $2
+-%C10-%C11-%O$tOffering %C11$1 %Cto %C11$2%O
 3
 
 DCC Offer Not Valid
 XP_TE_DCCIVAL
 pevt_generic_none_help
-%C22*%O$tNo such DCC offer.
+-%C10-%C11-%O$tNo such DCC offer.
 0
 
 DCC RECV Abort
 XP_TE_DCCRECVABORT
 pevt_dccfileabort_help
-%C22*%O$tDCC RECV%C26 $2%O to%C26 $1%O aborted.
+-%C10-%C11-%O$tDCC RECV %C11$2%O to %C11$1%O aborted.
 2
 
 DCC RECV Complete
 XP_TE_DCCRECVCOMP
 pevt_dccrecvcomp_help
-%C22*%O$tDCC RECV%C26 $1%O from%C26 $3%O complete %C30[%C26$4%O cps%C30]%O.
+-%C10-%C11-%O$tDCC RECV %C11$1%O from %C11$3%O complete %C14[%C11$4%O cps%C14]%O.
 4
 
 DCC RECV Connect
 XP_TE_DCCCONRECV
 pevt_dcccon_help
-%C22*%O$tDCC RECV connection established to%C26 $1 %C30[%O$2%C30]
+-%C10-%C11-%O$tDCC RECV connection established to %C11$1 %C14[%O$2%C14]%O
 3
 
 DCC RECV Failed
 XP_TE_DCCRECVERR
 pevt_dccrecverr_help
-%C22*%O$tDCC RECV%C26 $1%O from%C26 $3%O failed ($4).
+-%C10-%C11-%O$tDCC RECV $1 ($2) failed. Connection to $3 lost.
 4
 
 DCC RECV File Open Error
 XP_TE_DCCFILEERR
 pevt_generic_file_help
-%C22*%O$tDCC RECV: Cannot open $1 for writing ($2).
+-%C10-%C11-%O$tDCC RECV: Cannot open $1 for writing ($2).
 2
 
 DCC Rename
 XP_TE_DCCRENAME
 pevt_dccrename_help
-%C22*%O$tThe file%C26 $1%C already exists, saving it as%C26 $2%O instead.
+-%C10-%C11-%O$tThe file %C11$1%C already exists, saving it as %C11$2%O instead.
 2
 
 DCC RESUME Request
 XP_TE_DCCRESUMEREQUEST
 pevt_dccresumeoffer_help
-%C22*%O$t%C26$1 %Ohas requested to resume%C26 $2 %Cfrom%C26 $3%C.
+-%C10-%C11-%O$t%C11$1 %Chas requested to resume %C11$2 %Cfrom %C11$3%C.
 3
 
 DCC SEND Abort
 XP_TE_DCCSENDABORT
 pevt_dccfileabort_help
-%C22*%O$tDCC SEND%C26 $2%O to%C26 $1%O aborted.
+-%C10-%C11-%O$tDCC SEND %C11$2%O to %C11$1%O aborted.
 2
 
 DCC SEND Complete
 XP_TE_DCCSENDCOMP
 pevt_dccsendcomp_help
-%C22*%O$tDCC SEND%C26 $1%O to%C26 $2%O complete %C30[%C26$3%O cps%C30]%O.
+-%C10-%C11-%O$tDCC SEND %C11$1%O to %C11$2%O complete %C14[%C11$3%O cps%C14]%O.
 3
 
 DCC SEND Connect
 XP_TE_DCCCONSEND
 pevt_dcccon_help
-%C22*%O$tDCC SEND connection established to%C26 $1 %C30[%O$2%C30]
+-%C10-%C11-%O$tDCC SEND connection established to %C11$1 %C14[%O$2%C14]%O
 3
 
 DCC SEND Failed
 XP_TE_DCCSENDFAIL
 pevt_dccsendfail_help
-%C22*%O$tDCC SEND%C26 $1%O to%C26 $2%O failed. $3
+-%C10-%C11-%O$tDCC SEND %C11$1%O failed. Connection to %C11$2%O lost.
 3
 
 DCC SEND Offer
 XP_TE_DCCSENDOFFER
 pevt_dccsendoffer_help
-%C22*%O$t%C26$1 %Ohas offered%C26 $2 %O(%C26$3 %Obytes)
+-%C10-%C11-%O$t%C11$1 %Chas offered %C11$2 %C(%C11$3 %Cbytes)
 4
 
 DCC Stall
 XP_TE_DCCSTALL
 pevt_dccstall_help
-%C22*%O$tDCC $1%C26 $2 %Oto%C26 $3 %Cstalled - aborting.
+-%C10-%C11-%O$tDCC $1 %C11$2 %Cto %C11$3 %Cstalled - aborting.
 3
 
 DCC Timeout
 XP_TE_DCCTOUT
 pevt_dccstall_help
-%C22*%O$tDCC $1%C26 $2 %Oto%C26 $3 %Otimed out - aborting.
+-%C10-%C11-%O$tDCC $1 %C11$2 %Cto %C11$3 %Ctimed out - aborting.
 3
 
 Delete Notify
 XP_TE_DELNOTIFY
 pevt_generic_nick_help
-%C22*%O$t$1 deleted from notify list.
+-%C10-%C11-%O$t$1 deleted from notify list.
 1
 
 Disconnected
 XP_TE_DISCON
 pevt_discon_help
-%C22*%O$tDisconnected ($1).
+-%C10-%C11-%O$tDisconnected ($1).
 1
 
 Found IP
 XP_TE_FOUNDIP
 pevt_foundip_help
-%C22*%O$tFound your IP: [$1]
+-%C10-%C11-%O$tFound your IP: [$1]
 1
 
 Generic Message
@@ -409,13 +409,13 @@
 Ignore Add
 XP_TE_IGNOREADD
 pevt_ignoreaddremove_help
-%O%C26$1%O added to ignore list.
+%O%C11$1%O added to ignore list.
 1
 
 Ignore Changed
 XP_TE_IGNORECHANGE
 pevt_ignoreaddremove_help
-Ignore on %C26$1%O changed.
+Ignore on %C11$1%O changed.
 1
 
 Ignore Footer
@@ -433,7 +433,7 @@
 Ignore Remove
 XP_TE_IGNOREREMOVE
 pevt_ignoreaddremove_help
-%O%C26$1%O removed from ignore list.
+%O%C11$1%O removed from ignore list.
 1
 
 Ignorelist Empty
@@ -445,97 +445,97 @@
 Invite
 XP_TE_INVITE
 pevt_generic_channel_help
-%C22*%O$tCannot join%C26 %B$1 %O(Channel is invite only).
+-%C10-%C11-%O$tCannot join%C11 %B$1 %O(Channel is invite only).
 1
 
 Invited
 XP_TE_INVITED
 pevt_invited_help
-%C22*%O$tYou have been invited to%C26 $1%O by%C26 $2%C (%C26$3%C)
+-%C10-%C11-%O$tYou have been invited to %C11$1%C by %C11$2%C (%C11$3%C)
 3
 
 Join
 XP_TE_JOIN
 pevt_join_help
-%C19*%O$t%C19%B$1 %B($3) has joined $2
+-%C10-%C11>%O$t%B$1%B %C14(%C10$3%C14)%C has joined $2
 3
 
 Keyword
 XP_TE_KEYWORD
 pevt_generic_channel_help
-%C22*%O$tCannot join%C26 %B$1 %O(Requires keyword).
+-%C10-%C11-%O$tCannot join%C11 %B$1 %O(Requires keyword).
 1
 
 Kick
 XP_TE_KICK
 pevt_kick_help
-%C21*%O$t%C21$1 has kicked $2 from $3 ($4%O%C21)
+<%C10-%C11-%O$t$1 has kicked $2 from $3 ($4%O)
 4
 
 Killed
 XP_TE_KILL
 pevt_kill_help
-%C22*%O$tYou have been killed by $1 ($2%O%C22)
+-%C10-%C11-%O$tYou have been killed by $1 ($2%O)
 2
 
 Message Send
 XP_TE_MSGSEND
 pevt_ctcpsend_help
-%C19>%O$1%C19<%O$t$2
+%C3>%O$1%C3<%O$t$2%O
 2
 
 Motd
 XP_TE_MOTD
 pevt_servertext_help
-%C16*%O$t$1%O
+-%C10-%C11-%O$t$1%O
 1
 
 MOTD Skipped
 XP_TE_MOTDSKIP
 pevt_generic_none_help
-%C22*%O$t%C22MOTD Skipped.
+-%C10-%C11-%O$tMOTD Skipped.
 0
 
 Nick Clash
 XP_TE_NICKCLASH
 pevt_nickclash_help
-%C22*%O$t$1 already in use. Retrying with $2...
+-%C10-%C11-%O$t$1 already in use. Retrying with $2..
 2
 
 Nick Failed
 XP_TE_NICKFAIL
 pevt_generic_none_help
-%C22*%O$tNickname already in use. Use /NICK to try another.
+-%C10-%C11-%O$tNickname already in use. Use /NICK to try another.
 0
 
 No DCC
 XP_TE_NODCC
 pevt_generic_none_help
-%C22*%O$tNo such DCC.
+-%C10-%C11-%O$tNo such DCC.
 0
 
 No Running Process
 XP_TE_NOCHILD
 pevt_generic_none_help
-%C22*%O$tNo process is currently running
+-%C10-%C11-%O$tNo process is currently running
 0
 
 Notice
 XP_TE_NOTICE
 pevt_notice_help
-%C28-%C29$1%C28-%O$t$2
+%C12-%C13$1%C12-%O$t$2%O
 2
 
 Notice Send
 XP_TE_NOTICESEND
 pevt_ctcpsend_help
-%C19>%O$1%C19<%O$t$2
+%C3>%O$1%C3<%O$t$2%O
 2
 
 Notify Empty
 XP_TE_NOTIFYEMPTY
 pevt_generic_none_help
-$tNotify list is empty.
+-%C10-%C11-%O$tNotify list is empty.
 0
 
 Notify Header
@@ -547,19 +547,19 @@
 Notify Number
 XP_TE_NOTIFYNUMBER
 pevt_notifynumber_help
-%C22*%O$t$1 users in notify list.
+-%C10-%C11-%O$t$1 users in notify list.
 1
 
 Notify Offline
 XP_TE_NOTIFYOFFLINE
 pevt_generic_nick_help
-%C22*%O$tNotify: $1 is offline ($2).
+-%C10-%C11-%O$tNotify: $1 is offline ($2).
 2
 
 Notify Online
 XP_TE_NOTIFYONLINE
 pevt_generic_nick_help
-%C22*%O$tNotify: $1 is online ($2).
+-%C10-%C11-%O$tNotify: $1 is online ($2).
 2
 
 Open Dialog
@@ -571,139 +571,139 @@
 Part
 XP_TE_PART
 pevt_part_help
-%C23*%O$t%C23$1 (%O%C23$2) has left $3
+<%C10-%C11-%O$t$1 %C14(%O$2%C14)%C has left $3
 3
 
 Part with Reason
 XP_TE_PARTREASON
 pevt_partreason_help
-%C23*%O$t%C23$1 (%O%C23$2) has left $3 (%O%C23%B%B$4%O%C23)
+<%C10-%C11-%O$t$1 %C14(%O$2%C14)%C has left $3 %C14(%O$4%C14)%O
 4
 
 Ping Reply
 XP_TE_PINGREP
 pevt_pingrep_help
-%C22*%O$tPing reply from $1: $2 second(s)
+-%C10-%C11-%O$tPing reply from $1 : $2 second(s)
 2
 
 Ping Timeout
 XP_TE_PINGTIMEOUT
 pevt_pingtimeout_help
-%C22*%O$tNo ping reply for $1 seconds, disconnecting.
+-%C10-%C11-%O$tNo ping reply for $1 seconds, disconnecting.
 1
 
 Private Message
 XP_TE_PRIVMSG
 pevt_privmsg_help
-%C28*%C29$3$1%C28*$t%O$2
+%C12*%C13$1%C12*$t%O$2%O
 3
 
 Private Message to Dialog
 XP_TE_DPRIVMSG
 pevt_dprivmsg_help
-%C18%B%B$3$1%O$t$2
+%C2<%O$1%C2>%O$t$2%O
 3
 
 Process Already Running
 XP_TE_ALREADYPROCESS
 pevt_generic_none_help
-%C22*%O$tA process is already running
+-%C10-%C11-%O$tA process is already running
 0
 
 Quit
 XP_TE_QUIT
 pevt_quit_help
-%C23*%O$t%C23$1 has quit (%O%C23%B%B$2%O%C23)
+<%C10-%C11-%O$t$1 has quit %C14(%O$2%O%C14)%O
 3
 
 Raw Modes
 XP_TE_RAWMODES
 pevt_rawmodes_help
-%C22*%O$t$1 sets modes%B %C30[%O$2%B%C30]
+-%C10-%C11-%O$t$1 sets modes%B %C14[%O$2%B%C14]%O
 2
 
 Receive Wallops
 XP_TE_WALLOPS
 pevt_dprivmsg_help
-%C28-%C29$1/Wallops%C28-%O$t$2
+%C12-%C13$1/Wallops%C12-%O$t$2%O
 2
 
 Resolving User
 XP_TE_RESOLVINGUSER
 pevt_resolvinguser_help
-%C22*%O$tLooking up IP number for%C26 $1%O...
+-%C10-%C11-%O$tLooking up IP number for%C11 $1%O..
 2
 
 Server Connected
 XP_TE_SERVERCONNECTED
 pevt_generic_none_help
-%C22*%O$t%C22Connected.
+-%C10-%C11-%O$tConnected.
 0
 
 Server Error
 XP_TE_SERVERERROR
 pevt_servererror_help
-%C22*%O$t$1
+-%C10-%C11-%O$t$1%O
 1
 
 Server Lookup
 XP_TE_SERVERLOOKUP
 pevt_serverlookup_help
-%C22*%O$t%C22Looking up $1
+-%C10-%C11-%O$tLooking up %C11$1%C..
 1
 
 Server Notice
 XP_TE_SERVNOTICE
 pevt_servertext_help
-%C22*%O$t$1
+-%C10-%C11-%O$t$1%O
 2
 
 Server Text
 XP_TE_SERVTEXT
 pevt_servertext_help
-%C22*%O$t$1
+-%C10-%C11-%O$t$1%O
 2
 
 Stop Connection
 XP_TE_STOPCONNECT
 pevt_sconnect_help
-%C22*%O$tStopped previous connection attempt (pid=$1)
+-%C10-%C11-%O$tStopped previous connection attempt (pid=$1)
 1
 
 Topic
 XP_TE_TOPIC
 pevt_topic_help
-%C29*%O$t%C29Topic for $1%C %C29is: $2
+-%C10-%C11-%O$tTopic for %C11$1%C is %C11$2%O
 2
 
 Topic Change
 XP_TE_NEWTOPIC
 pevt_newtopic_help
-%C22*%O$t$1 has changed the topic to: $2
+-%C10-%C11-%O$t$1 has changed the topic to: $2%O
 3
 
 Topic Creation
 XP_TE_TOPICDATE
 pevt_topicdate_help
-%C29*%O$t%C29Topic for $1%C %C29set by $2%C %C29at $3
+-%C10-%C11-%O$tTopic for %C11$1%C set by %C11$2%C at %C11$3%O
 3
 
 Unknown Host
 XP_TE_UKNHOST
 pevt_generic_none_help
-%C22*%O$tUnknown host. Maybe you misspelled it?
+-%C10-%C11-%O$tUnknown host. Maybe you misspelled it?
 0
 
 User Limit
 XP_TE_USERLIMIT
 pevt_generic_channel_help
-%C22*%O$tCannot join%C26 %B$1 %O(User limit reached).
+-%C10-%C11-%O$tCannot join%C11 %B$1 %O(User limit reached).
 1
 
 Users On Channel
 XP_TE_USERSONCHAN
 pevt_usersonchan_help
-%C22*%O$t%C26Users on $1:%C $2
+-%C10-%C11-%O$t%C11Users on $1:%C $2
 2
 
 WhoIs Authenticated
@@ -715,55 +715,55 @@
 WhoIs Away Line
 XP_TE_WHOIS5
 pevt_whois5_help
-%C22*%O$t%C28[%O$1%C28] %Cis away %C30(%O$2%O%C30)
+-%C10-%C11-%O$t%C12[%O$1%C12] %Cis away %C14(%O$2%O%C14)
 2
 
 WhoIs Channel/Oper Line
 XP_TE_WHOIS2
 pevt_whois2_help
-%C22*%O$t%C28[%O$1%C28]%O $2
+-%C10-%C11-%O$t%C12[%O$1%C12]%C $2
 2
 
 WhoIs End
 XP_TE_WHOIS6
 pevt_whois6_help
-%C22*%O$t%C28[%O$1%C28] %OEnd of WHOIS list.
+-%C10-%C11-%O$t%C12[%O$1%C12] %CEnd of WHOIS list.
 1
 
 WhoIs Identified
 XP_TE_WHOIS_ID
 pevt_whoisid_help
-%C22*%O$t%C28[%O$1%C28]%O $2
+-%C10-%C11-%O$t%C12[%O$1%C12] %C$2
 2
 
 WhoIs Idle Line
 XP_TE_WHOIS4
 pevt_whois4_help
-%C22*%O$t%C28[%O$1%C28]%O idle%C26 $2
+-%C10-%C11-%O$t%C12[%O$1%C12]%O idle %C11$2%O
 2
 
 WhoIs Idle Line with Signon
 XP_TE_WHOIS4T
 pevt_whois4t_help
-%C22*%O$t%C28[%O$1%C28]%O idle%C26 $2%O, signon:%C26 $3
+-%C10-%C11-%O$t%C12[%O$1%C12]%O idle %C11$2%O, signon: %C11$3%O
 3
 
 WhoIs Name Line
 XP_TE_WHOIS1
 pevt_whois1_help
-%C22*%O$t%C28[%O$1%C28] %C30(%O$2@$3%C30)%O: $4
+-%C10-%C11-%O$t%C12[%O$1%C12] %C14(%O$2@$3%C14) %O: $4%O
 4
 
 WhoIs Real Host
 XP_TE_WHOIS_REALHOST
 pevt_whoisrealhost_help
-%C22*%O$t%C28[%O$1%C28] %Oreal user@host%C27 $2%O, real IP%C27 $3
+-%C10-%C11-%O$t%C12[%O$1%C12] %Oreal user@host %C11$2%O, real IP %C11$3%O
 4
 
 WhoIs Server Line
 XP_TE_WHOIS3
 pevt_whois3_help
-%C22*%O$t%C28[%O$1%C28]%O $2
+-%C10-%C11-%O$t%C12[%O$1%C12]%O $2
 2
 
 WhoIs Special
@@ -773,44 +773,44 @@
 3
 
 You Join
-XP_TE_UJOIN
-pevt_join_help
-%C19*%O$t%C19Now talking on $2
-3
+XP_TE_UKICK
+pevt_ukick_help
+-%C10-%C11>%O$t%BYou%B are now talking on %C11$2%O
+4
 
 You Kicked
 XP_TE_UKICK
 pevt_ukick_help
-%C23*$tYou have been kicked from $2 by $3 ($4%O%C23)
+-%C10-%C11-%O$tYou have been kicked from $2 by $3 ($4%O)
 4
 
 You Part
 XP_TE_UPART
 pevt_part_help
-%C23*$tYou have left channel $3
+-%C10-%C11-%O$tYou have left channel $3
 3
 
 You Part with Reason
 XP_TE_UPARTREASON
 pevt_partreason_help
-%C23*$tYou have left channel $3 (%O%C23%B%B$4%O%C23)
+-%C10-%C11-%O$tYou have left channel $3 %C14(%O$4%C14)%O
 4
 
 Your Invitation
 XP_TE_UINVITE
 pevt_uinvite_help
-%C22*%O$tYou've invited%C26 $1%O to%C26 $2%O (%C26$3%O)
+-%C10-%C11-%O$tYou're inviting %C11$1%C to %C11$2%C (%C11$3%C)
 3
 
 Your Message
 XP_TE_UCHANMSG
 pevt_chanmsg_help
-%C31%B%B$4$1%O%C30$t$2
+%C6<%O$1%C6>%O$t$2%O
 4
 
 Your Nick Changing
 XP_TE_UCHANGENICK
 pevt_uchangenick_help
-%C22*%O$tYou are now known as $2
+-%C10-%C11-%O$tYou are now known as $2
 2
 
Index: src/pixmaps/Makefile.am
===================================================================
RCS file: /cvsroot/xchat/xchat2/src/pixmaps/Makefile.am,v
retrieving revision 1.8
diff -u -r1.8 Makefile.am
--- src/pixmaps/Makefile.am	9 Jan 2003 06:22:30 -0000	1.8
+++ src/pixmaps/Makefile.am	10 Sep 2005 22:03:39 -0000
@@ -14,5 +14,8 @@
 CLEANFILES = $(noinst_HEADERS)
 EXTRA_DIST = $(PNGS) makefile.msc
 
+xchatdatadir = $(prefix)/share/xchat
+xchatdata_DATA = $(PNGS)
+
 inline_pngs.h:
 	@gdkpixbufcsourcepath@ --build-list $(LIST) > $(srcdir)/inline_pngs.h
