--- snes9x-1.42-src.orig/snes9x/unix/x11.cpp	2003-12-04 10:03:08.000000000 -0700
+++ snes9x-1.42-src/snes9x/unix/x11.cpp	2004-09-14 21:08:42.492777936 -0600
@@ -273,8 +273,8 @@
     GUI.screen = DefaultScreenOfDisplay (GUI.display);
     GUI.screen_num = XScreenNumberOfScreen (GUI.screen);
     GUI.visual = DefaultVisualOfScreen (GUI.screen);
-    GUI.window_width = IMAGE_WIDTH;
-    GUI.window_height = IMAGE_HEIGHT;
+    GUI.window_width = 800;
+    GUI.window_height = 600;
 
 #ifdef USE_OPENGL
     // XXX:
